src/main.rs

```diff
@@ -130,6 +131,23 @@ fn main() -> anyhow::Result<()> {
         return Ok(());
     }
 
+    let on_exit = || match crossterm::terminal::disable_raw_mode() {
```

Comment:
since this is context free, you can also specify a full `fn` in a `fn` body :)

Reply:
Do we have to check if we are in raw mode first?
Or maybe even track the initial terminal state and restore on exit?

Reply:
`crossterm` has no way to query the mode, just enable or disable. Reset to initial state is a good idea, I'll have a look at that!